resources/metadata/xsDeploy.yaml

```diff
@@ -0,0 +1,128 @@
+metadata:
+  name: xsDeploy
+  description: Performs xs deployment
+  longDescription: |
+    Performs xs deployment
+
+spec:
+  inputs:
+    resources:
+    params:
+      - name: deployOpts
+        type: string
+        description: Additional deploy options
+        scope:
+        - GENERAL
+        - PARAMETERS
+        - STAGES
+        - STEPS
+        mandatory: false
+        default:
+      - name: mtaPath
+        type: string
+        description: Path to deployable
+        scope:
+        - GENERAL
+        - PARAMETERS
+        - STAGES
+        - STEPS
+        mandatory: true
+      - name: action
+        type: string
+        description: The action
+        default: None
+        scope:
+        - GENERAL
+        - PARAMETERS
+        - STAGES
+        - STEPS
+        mandatory: false
+      - name: mode
+        type: string
+        description: The mode
+        default: Deploy
+        scope:
+        - GENERAL
+        - PARAMETERS
+        - STAGES
+        - STEPS
+        mandatory: true
+      - name: deploymentID
+        type: string
+        description: The deployment ID
+        default:
+        scope:
+        - GENERAL
+        - PARAMETERS
+        - STAGES
+        - STEPS
+        mandatory: false
+      - name: apiUrl
+        type: string
+        description: The api url (e.g. https://example.org:12345
+        scope:
+        - GENERAL
+        - PARAMETERS
+        - STAGES
+        - STEPS
+        mandatory: true
+      - name: user
+        type: string
+        description: User
+        scope:
+        - GENERAL
```

Comment:
not sure if GENERAL really makes sense for this parameter, I suggest to use it only selectively ...